0004-post-tags.md

```diff
@@ -0,0 +1,299 @@
+Shared parameters would be authentication, `post_id` and `tag_id` with an additional `delete` parameter for the delete endpoint.
+
+Groups that should be able to add/edit the tags:
+- Instance Admins of the Community
+- Moderators of the Community
+- Post creator
+
+### Frontend:
+
+**General Changes:**
+
+Tags can be listed next to the Post title in the feed and at the end of the post in the post view.
+Adding tags to a post can be implemented via a separate tag box where they can be typed in with search support. 
+
+![tags post view iamge](https://github.com/Neshura87/Lemmy-RFC/blob/main/tags_post_view.png?raw=true)
+
+Tags in the feed should be displayed next to the post interaction elements.
+![tags feed view iamge](https://github.com/Neshura87/Lemmy-RFC/blob/main/tags_feed_view.png?raw=true)
+
+
+
+**Moderator Changes:**
+
+Moderators need a UI Section in the community/instance settings for adding, edditing and deleting tags. This could also implement a way to "import" a tag from another instance.
+Moderators should also be able to add tags to user posts.
+
+# Drawbacks
+
+Adding the tag system would increase backend complexity, increased workload for alternative frontends and apps as well as add another barrier for full federation between lemmy and other services. On top of that the moderation workload would somewhat increase since posts would need to be checked for proper usage of tags as well.
+
+# Rationale and alternatives
+
+The proposed system aims to be a balance between the extremely limited flair feature of reddit and the often fragmented tag system of image hosting platforms, combining the best properties of both options into a theoretically ideal version of the system. Additionally the current system with a singular NSFW flag is far too inflexible to properly moderate the scope of content that falls under NSFW. Many instances have implemented a blanket NSFW ban because of this, greatly limiting what content can be posted where. Aside from that the inability to better sort content is a often discussed topic, especially story communities are desperately craving for a better option to both sort posts based on story association as well as hide/blur posts if they contain spoilers. Given the very high community demand not implementing a tag system of some sort will likely lead to a (soft-)fork of lemmy. The feature is extremely desired by many communities. An implementation of tags via UI rendering ([] brackets for tags in text would turn into a tag on a post) would be possible but ultimately far more limiting and would lead to an incosistent experience across lemmy apps/frontends as well as an inability to properly federate the tags.
+
+# Prior art
+
+Similar implementations can be found on Reddit as well as a great many image hosting platforms.
+
+Reddit's Flair System is rather restrictive, a Post can only have a single custom Flair (such as "News") greatly limiting the usefullness of the feature.
```

Comment:
There are also cases where multiple tags dont make sense, for example there shouldnt be a post tagged as "Meme" and "News". So it would make some sense to follow Reddit on this one. This would also make the database changes much easier.

Reply:
Mh here I think allowing multiple tags should be there from the start

Reply:
Absolutely do not want reddit's system as it is way too restrictive to really make use of tags but I think a limit for tags can be implemented on a community level for moderators and admins to regulate. Since I have a hard time judging how much work that would add to the RFC I've added it to the Future possibilities section, if it's deemed to important to ship later I'll move it up somewhere into the text. (see 5a3082a01fbd9df1d07beab2141cef71a3972c0b)